src/Main.hs

```diff
@@ -106,9 +104,35 @@ logProgress args s = case Args.outputAction args of
   AnalysisAction CompilerOutput -> hPutStrLn stdout s
   GraphAction -> hPutStrLn stderr s
 
-dumpCallGraph :: Args.Args -> Config -> [CTranslUnit] -> IO ()
+dumpCallGraph :: Args.Args -> Config -> [ProcessingUnit] -> IO ()
 dumpCallGraph args config translationUnits = do
-  let callMap = Graph.fromTranslationUnits config
+  let callMap = Graph.fromProcessingUnits config
         (zip (Args.translationUnitPaths args) translationUnits)
   DumpCallMap.hPutCallMap stdout callMap
 
+parseInput :: Args.Args -> FilePath -> IO (Either ProcessingUnitParseError ProcessingUnit)
+parseInput args path =
+  case classifyPath path of
+    CSourcePathClass ->
+      (either (Left . CSourceUnitParseError) (Right . CSourceProcessingUnit)) <$> parseCInput args path
```

Comment:
`either (Left . f) (Right . g)` = `bimap f g`. This pattern shows up in a few places.